### DISCUSSION TITLE
Log view w/o scrolling

### DISCUSSION THREAD

**Discussion by frosch01** [zauberzeug/nicegui#2690] [Ideas / Feature Requests]
# Log view scrolling

The log view is a simple and cool way for offering debugging from an application. I also added it in my project. Thx a lot for this cool element.

## Jumping to the end can be annoying

As new lines are added to the log, the scroll-bar also moves to the end. This is cool to follow the logs. However, if there is something unexpected found while observing the logs, I want to stop this automatic jumping and instead fix the scroll-bar at the current position. When I am done with analyzing, I move "back" to the end of the log.

E.g., on Linux terminals, if the scroll bar is not at it's lower end, the view will stay where it is without jumping to the end. This is intuitive and cool, but having a checkbox for this one should do the same trick.

## Disabling the jumping

I tried to understand the code a bit. To me, it seems there is some code that sends the scroll-bar to the end:

```textarea.scrollTop = textarea.scrollHeight;```

## Failed to hack myself

I wanted to check the behavior when this line is commented, but I was not able to ship the changed log.js to the browser. I had the idea that it is picked as found in the sandbox but at the end, it seems the file as found in version 1.4.9 was shipped to my browser. 

I am really too far from web development and it's basic architectures. So, if you think there is a chance that I will be successful, perhaps you can give me a tip on how to hack myself.

What I would expect from this experiment is that there most likely is still some scrolling as the position of the the scroll bar needs to change as new text is added, too. Well, perhaps things are already solved implicitly.

## Is this a general interest?

If this is interesting for others, perhaps on of the core developers could help out. I feel that things should actually not be too complicated.

**Comment by falkoschindler** [comment]
That's a great idea, @frosch01!
Actually I'm working on a new `ui.log` element (see PR #2686), where I already implemented your suggestion. It turned out to be pretty easy to get such a dynamic auto-scroll using some CSS properties like `scroll-snap-type` and `scroll-snap-align`.

**Reply by frosch01** [reply]
Thx a lot for providing a solution so fast. What a great response time.
